source/common/router/rds_impl.cc

```diff
@@ -146,30 +147,42 @@ RouteConfigProviderManagerImpl::RouteConfigProviderManagerImpl(
     const LocalInfo::LocalInfo& local_info, ThreadLocal::SlotAllocator& tls, Server::Admin& admin)
     : runtime_(runtime), dispatcher_(dispatcher), random_(random), local_info_(local_info),
       tls_(tls), admin_(admin) {
-  admin_.addHandler("/routes", "print out currently loaded dynamic HTTP route tables",
-                    MAKE_ADMIN_HANDLER(RouteConfigProviderManagerImpl::handlerRoutes), true, false);
+  config_tracker_entry_ =
+      admin_.getConfigTracker().add("routes", [this] { return dumpRouteConfigs(); });
+  // ConfigTracker keys must be unique. We are asserting that no one has stolen the "routes" key
+  // from us, since the returned entry will be nullptr if the key already exists.
+  RELEASE_ASSERT(config_tracker_entry_);
```

Comment:
I think this falls into the `ASSERT` category, it's an internal programming invariant rather than a statement about the runtime system. `RELEASE_ASSERT` is largely used for unrecoverable environmental issues such as OOM or some unexpected syscall behavior that should "never happen".

Reply:
@htuch FWIW I recommended using `RELEASE_ASSERT` here because it's theoretically possible for a plugin to register a config tracker entry with the name "routes" which would be pretty confusing. This would at least crash in all builds. I don't feel very strongly either way.

Reply:
Fair enough, I would buy the argument that a plugin is an external environment factor that we can't control for as an internal invariant.